src/library_async.js

```diff
@@ -410,6 +410,101 @@ mergeInto(LibraryManager.library, {
     });
   },
 
+  /*
+   * Layout of an EMTERPRETIFY_ASYNC coroutine structure:
+   *
+   *  0 callee's EMTSTACKTOP
```

Comment:
I don't think we need to track the global var `EMTSTACKTOP` etc. - we do set it up in JS, but then responsibility is in the compiled code. So we just need that.

Reply:
`EMTSTACKTOP` is used at least [here](https://github.com/atrosinenko/emscripten/blob/b8e4f8c3cfe1e9b043bd4213ad9a960eed4fcd7e/src/library_async.js#L265). And so [here](https://github.com/atrosinenko/emscripten/blob/b8e4f8c3cfe1e9b043bd4213ad9a960eed4fcd7e/src/library_async.js#L309) it is used to find the first stack frame (I use it likewise in my PR) -- as I understand, it is the global var, not one from the compiled code. Meanwhile, if it is used only for initialization, it would be great to rename it like `INITIAL_EMTSTACKTOP` -- it took some time for me to understand why one creates subarray from `EMTSTACKTOP` to `emtStackSave()` that returns `EMTSTACKTOP` too. :) But it is another story, of course.

Reply:
I see, thanks. Looks like it's used as the very initial start of the stack, yeah. During startup, we should copy it to `EMTSTACK_BASE` and use that (we do a similar thing for the normal stack), and otherwise never use `EMTSTACKTOP` in JS.